tests/ignite/engine/test_create_supervised.py

```diff
@@ -87,6 +87,40 @@ def _test_create_supervised_trainer(
                 trainer.run(data)
 
 
+def test_create_supervised_training_scalar_assignment():
+    model = Linear(1, 1)
+
+    model.weight.data.zero_()
+    model.bias.data.zero_()
+    optimizer = SGD(model.parameters(), 0.1)
+
+    with mock.patch("ignite.engine._check_arg") as check_arg_mock:
+        check_arg_mock.return_value = None, torch.cuda.amp.GradScaler(enabled=True)
+        trainer = create_supervised_trainer(
+            model,
+            optimizer,
+            mse_loss,
+            device="cpu",
+            output_transform=lambda x, y, y_pred, loss: (y_pred, loss.item()),
+            amp_mode=None,
+            scaler=True,
+        )
+        assert hasattr(trainer.state, "scaler")
+        assert isinstance(trainer.state.scaler, torch.cuda.amp.GradScaler)
+
+        check_arg_mock.return_value = None, None
+        trainer = create_supervised_trainer(
+            model,
+            optimizer,
+            mse_loss,
+            device="cpu",
+            output_transform=lambda x, y, y_pred, loss: (y_pred, loss.item()),
+            amp_mode=None,
+            scaler=True,
+        )
+        assert not hasattr(trainer.state, "scaler")
```

Comment:
Which part of code we test here ?

Reply:
@vfdev-5 I wanted my test to be exhaustive, so I tested the two possible scenarios of the trainer state assignment, when it was set to a grad scalar and when it wasn't set.

Reply:
Yes, I understand that but the config as `amp_mode=None, scaler=True` looks very strange...

Reply:
That makes sense, I will remove the second test case then